### ISSUE TITLE
Decompilation via IDA plugin fails with "FileNotFoundError: The system cannot find the file specified"

### ISSUE THREAD

**Issue by E4ck** [avast-tl/retdec#434]
launch_process(F:\Python37\python.exe "F:\retdec-3.2\scripts\retdec-decompiler.py" "F:\baby_mips" --config="F:\baby_mips.idb.json" --color-for-ida -o "F:\baby_mips.c" --select-decode-only --select-ranges="0x4106b0-0x410757") failed with error code 1

If you can give me an email address, I can send the binary to you. thanks

**Comment by E4ck** [comment]
> launch_process(F:\Python37\python.exe "F:\retdec-3.2\scripts\retdec-decompiler.py" "F:\baby_mips" --config="F:\baby_mips.idb.json" --color-for-ida -o "F:\baby_mips.c" --select-decode-only --select-ranges="0x4106b0-0x410757") failed with error code 1
> 
> If you can give me an email address, I can send the binary to you. thanks

After confirming this error, another prompt will pop up: Loading of output C file FAILED.

**Comment by E4ck** [comment]
> launch_process(F:\Python37\python.exe "F:\retdec-3.2\scripts\retdec-decompiler.py" "F:\baby_mips" --config="F:\baby_mips.idb.json" --color-for-ida -o "F:\baby_mips.c" --select-decode-only --select-ranges="0x4106b0-0x410757") failed with error code 1
> 
> If you can give me an email address, I can send the binary to you. thanks

After confirming this error, another prompt will pop up: Loading of output C file FAILED.

I am using these versions:
ida: 7.0
python: 3.7.1

**Comment by PeterMatula** [comment]
Few questions:
- Are you using RetDec IDA plugin?
- You have focus in some function and trigger the decompilation?
- Are you able to decompile some function from some other binary via IDA plugin?
- Are you able to decompile some other function from this binary (`F:\baby_mips`) via IDA plugin?
- Are you able to decompile some other binary via `retdec-decompiler.py` (not IDA plugin)?
- Are you able to decompile this binary (`F:\baby_mips`) via `retdec-decompiler.py` (not IDA plugin)?

Basically, is this problem specific to this binary and/or IDA plugin, or does it always happen?

You can send the binary to info@retdec.com. Please reference this issue somewhere in the email. If the problem happens only for some specific function, mention which one. I can try to decompile it myself.

**Comment by E4ck** [comment]
> 几个问题：
> 
> * 你在使用RetDec IDA插件吗？
> * 你有专注于某些功能并触发反编译？
> * 你能通过IDA插件从其他二进制文件中反编译一些函数吗？
> * 你`F:\baby_mips`能通过IDA插件从这个二进制文件（）中反编译一些其他函数吗？
> * 您是否能够通过`retdec-decompiler.py`（而不是IDA插件）反编译其他二进制文件？
> * 你`F:\baby_mips`能通过`retdec-decompiler.py`（而不是IDA插件）反编译这个二进制文件（）吗？
> 
> 基本上，这个二进制和/或IDA插件特有的问题，还是总是发生？
> 
> 您可以将二进制文件发送到[info@retdec.com](mailto:info@retdec.com)。请在电子邮件中的某处参考此问题。如果问题仅针对某些特定功能发生，请提及哪一个。我可以尝试自己反编译。

Hi, thank you for your reply, I have sent the binary to your mailbox.
For your question: I am having this problem when using retdec-idaplugin. I tried several binary files and I encountered this problem, and it is the same situation to disassemble any function with Ctrl+D.
So, I guess what is wrong with my installation operation, this is my environment variable:
F:\retdec-3.2\scripts\
F:\Python37\Scripts\
F:\Python37\
At the same time, I still have a python2 environment, but I set it to be python3 when I use python in the terminal.
I only set the Retdec plguin option in ida when I use it, and then disassemble it by ctrl+D.

**Comment by E4ck** [comment]
I forgot to say that under Windows I will use retdec-decompiler.py to decompile baby_mips and report an error.
##### Checking if file is a Mach-O Universal static library...
Traceback (most recent call last):
  File "F:\retdec-3.2\scripts\retdec-decompiler.py", line 1275, in <module>
    sys.exit(decompiler.decompile())
  File "F:\retdec-3.2\scripts\retdec-decompiler.py", line 653, in decompile
    if utils.is_macho_archive(self.input_file):
  File "F:\retdec-3.2\scripts\retdec-utils.py", line 387, in is_macho_archive
    _, ret, _ = CmdRunner().run_cmd([config.EXTRACT, '--check-archive', path], discard_stdout=True, discard_stderr=True)
  File "F:\retdec-3.2\scripts\retdec-utils.py", line 55, in run_cmd
    _, output, return_code, timeouted = self._run_cmd(cmd, input, timeout, buffer_output, track_memory=False, discard_stdout=discard_stdout, discard_stderr=discard_stderr, print_run_msg=print_run_msg)
  File "F:\retdec-3.2\scripts\retdec-utils.py", line 88, in _run_cmd
    p = self._start(cmd, buffer_output, discard_stdout=discard_stdout, discard_stderr=discard_stderr)
  File "F:\retdec-3.2\scripts\retdec-utils.py", line 163, in _start
    return _WindowsProcess(**kwargs)
  File "F:\retdec-3.2\scripts\retdec-utils.py", line 249, in __init__
    super().__init__(**kwargs)
  File "F:\Python37\lib\subprocess.py", line 769, in __init__
    restore_signals, start_new_session)
  File "F:\Python37\lib\subprocess.py", line 1172, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] The system cannot find the file specified.

**Comment by E4ck** [comment]
Hi, thank you for your reply, I have sent the binary to your mailbox.

For your question: I am having this problem when using retdec-idaplugin. I tried several binary files and I encountered this problem, and it is the same situation to disassemble any function with Ctrl+D.

So, I guess what is wrong with my installation operation, this is my environment variable:

F:\retdec-3.2\scripts

F:\Python37\Scripts

F:\Python37

At the same time, I still have a python2 environment, but I set it to be python3 when I use python in the terminal.

I only set the Retdec plguin option in ida when I use it, and then disassemble it by ctrl+D.

I use retdec-decompiler.py (not the IDA plugin) to disassemble baby_mips on windows and report an error:

Checking if file is a Mach-O Universal static library...

Traceback (most recent call last):

File "F:\retdec-3.2\scripts\retdec-decompiler.py", line 1275, in 

sys.exit(decompiler.decompile())

File "F:\retdec-3.2\scripts\retdec-decompiler.py", line 653, in decompile

if utils.is_macho_archive(self.input_file):

File "F:\retdec-3.2\scripts\retdec-utils.py", line 387, in is_macho_archive

_, ret, _ = CmdRunner().run_cmd([config.EXTRACT, '--check-archive', path], discard_stdout=True, discard_stderr=True)

File "F:\retdec-3.2\scripts\retdec-utils.py", line 55, in run_cmd

, output, return_code, timeouted = self.run_cmd(cmd, input, timeout, buffer_output, track_memory=False, discard_stdout=discard_stdout, discard_stderr=discard_stderr, print_run_msg=print_run_msg)

File "F:\retdec-3.2\scripts\retdec-utils.py", line 88, in _run_cmd

p = self._start(cmd, buffer_output, discard_stdout=discard_stdout, discard_stderr=discard_stderr)

File "F:\retdec-3.2\scripts\retdec-utils.py", line 163, in _start

return _WindowsProcess(**kwargs)

File "F:\retdec-3.2\scripts\retdec-utils.py", line 249, in init

super().init(**kwargs)

File "F:\Python37\lib\subprocess.py", line 769, in init

restore_signals, start_new_session)

File "F:\Python37\lib\subprocess.py", line 1172, in _execute_child

startupinfo)

FileNotFoundError: [WinError 2] The system cannot find the file specified.

**Comment by s3rvac** [comment]
Based on this line from the output you have posted
```
File "F:\retdec-3.2\scripts\retdec-decompiler.py", line 653, in decompile
```
it seems to me that your are trying to run RetDec from its source directory. You need to install RetDec and run it from this installed directory (e.g. `F:\retdec-3.2-install\bin\retdec-decompiler.py`). On Windows, your can download, unpack, and use [this release](https://github.com/avast-tl/retdec/releases/download/v3.2/retdec-v3.2-windows-64b.zip) so you do not have to build RetDec by yourself.

**Comment by E4ck** [comment]
Thank you for your answer, but I tried to download retdec-v3.2-windows-64b.zip before, no success, google chrome shows failure - has been banned, is github have permission settings?

**Comment by s3rvac** [comment]
No idea. We have no problems with downloading it via Chrome. Also, VirusTotal shows no AV detections ([URL](https://www.virustotal.com/#/url/1cbdf91ae6c57c73f9e88cb10b0d8e801c0f3058d24d4a725443b257ad238be7/detection), [archive](https://www.virustotal.com/#/file/8f66cd4ae4200dc8c2d7aa12891f00299ba35ac6c3adf33ac23243af1498d1de/detection)), so I see no reason why it should not work. Could you try another browser?

**Comment by E4ck** [comment]
Thank you, I solved my problem according to your suggestion.
